### PR TITLE
[SG-35283]: Wildcard V2: Apply <Input /> Codemod `auth` `components` `extensions` `marketing` `notebooks`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Problem statement

Migration is required to update our code to use the `<Input />` Wildcard component.
See [parent issue](https://github.com/sourcegraph/sourcegraph/issues/27722) for more context. 

## Success criteria

All occurrences of the `<input />` element are replaced with the `<Input />` Wildcard component.

## Implementation details

- Rename the `Input` component `variant` prop to `size`.
- Use the `Input` codemod created [here](https://github.com/sourcegraph/codemod/pull/94) to speed up the migration.

## Refs
[Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/35283)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-35283)

## Affected Packages includes
- `client/web/src/auth`
- `client/web/src/components`
- `client/web/src/extensions`
- `client/web/src/marketing`
- `client/web/src/notebooks`

## Test Plan
- All `<input/>` elements from the mentioned folders are migrated to `<Input/>` wildcard component.
- No UI diff available

## App preview:

- [Web](https://sg-web-contractors-sg-35283.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-hijismaywt.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by valerybugakov — CHANGES_REQUESTED** [review]
Let's remove all the redundant changes from PRs before requesting the review.

File: client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
+:root {
+    --input-placeholder-color: #757575;
```
Comment (valerybugakov) [inline client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss:2]: Could you share the reasoning behind this change?

File: client/wildcard/src/components/Form/Input/Input.tsx [hunk 2]
```diff
@@ -70,11 +70,17 @@ export const Input = forwardRef((props, reference) => {
...
                     disabled={disabled}
                     type={type}
-                    className={classNames(styles.input, inputClassName, 'form-control', 'with-invalid-icon', {
-                        'is-valid': status === InputStatus.valid,
-                        'is-invalid': error || status === InputStatus.error,
-                        'form-control-sm': variant === 'small',
-                    })}
+                    className={classNames(
+                        inputClassName,
+                        status === InputStatus.loading && styles.inputLoading,
```
Comment (valerybugakov) [inline client/wildcard/src/components/Form/Input/Input.tsx:75]: Let's merge the `main` branch into this PR.

File: client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.input {
+    display: none !important;
```
Comment (valerybugakov) [inline client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss:2]: Do we need `!important` here?

File: client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx [hunk 4]
```diff
@@ -107,21 +106,23 @@ export const NotebookFileBlockInputs: React.FunctionComponent<
...
+                <Input
                     id={`${id}-line-range-input`}
-                    type="text"
-                    className={classNames('form-control', isLineRangeValid === false && 'is-invalid')}
+                    status={InputStatus[isLineRangeValid === false ? 'error' : 'initial']}
                     value={lineRangeInput}
                     onChange={onLineRangeInputChange}
                     placeholder="Enter a single line (1), a line range (1-10), or leave empty to show the entire file."
+                    label="Line range"
+                    className={isLineRangeValid === false ? 'mb-0' : 'mb-0'}
```
Comment (valerybugakov) [inline client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx:116]: Should we convert it to `mb-0` without ternary then?

File: client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx [hunk 5]
```diff
@@ -107,21 +106,23 @@ export const NotebookFileBlockInputs: React.FunctionComponent<
...
+                    error={
+                        isLineRangeValid === false && (
+                            <span className={styles.inputError}>
```
Comment (valerybugakov) [inline client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx:119]: Is this `className` required only to change the error font-size from the default one to the custom one? Does it make sense to keep it consistent with every other `Input` error in the application?

**Review by valerybugakov — CHANGES_REQUESTED** [review]
Let's fix the Chromatic diff shown for the [CollaboratorsPanel](https://www.chromatic.com/component?appId=5f0f381c0e50750022dc6bf7&csfId=web-search-panels-collaboratorspanel&specName=CollaboratorsPanel&buildNumber=43993) story. It seems that input borders shouldn't be rounded on the right side.

**Comment by gitstart-sourcegraph** [comment]
Hey @valerybugakov apologies this PR is still in progress, needed to mark it as ready for review in order to get chromatic review CI. Would update requested changes and request another review once ready.

File: client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss (hunk 3 above)
Comment (gitstart-sourcegraph) [reply client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss:2]: it's needed to override `d-flex` (has `important`) property applied internally (in `Input` component) on `LoaderInput`.

File: client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx (hunk 5 above)
Comment (gitstart-sourcegraph) [reply client/web/src/notebooks/blocks/file/NotebookFileBlockInputs.tsx:119]: Yea makes sense

File: client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss:2]: This was added to resolve UI diff on placeholder input color here.

Default (Before updating placeholder color with this change)
Screenshot 2022-05-30 at 7 05 28 AM

Expected (Existing placeholder color)
Screenshot 2022-05-30 at 7 02 11 AM

WDYT?

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, got it! Makes sense. We can convert back to draft after triggering the Chromatic build.

File: client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/notebooks/notebook/NotebookCommandPaletteInput.module.scss:2]: I think we can keep the default color here for consistency and better contrast reasons.

File: .github/workflows/lsif-typescript.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+name: lsif-typescript
```
Comment (valerybugakov) [inline .github/workflows/lsif-typescript.yml:1]: Should we revert this change?

File: .github/workflows/reviewdog.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+name: reviewdog
```
Comment (valerybugakov) [inline .github/workflows/reviewdog.yml:1]: Should we revert this change?

File: client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.input {
+    display: none !important; // Overrides d-flex css class on Input;
```
Comment (valerybugakov) [inline client/web/src/notebooks/listPage/ImportMarkdownNotebookButton.module.scss:2]: Let's convert the `d-flex` class in the `Input` component to a CSS module class. Then the `!important` here will become redundant.

**Review by valerybugakov — CHANGES_REQUESTED** [review]

**Review by valerybugakov — APPROVED** [review]
Thanks!

**Pull request merged**